compiler/noirc_evaluator/src/acir/tests/instructions/binary.rs

```diff
@@ -0,0 +1,631 @@
+    BRILLIG CALL func: 0, inputs: [w3], outputs: [w4]
+    ASSERT w5 = -w3*w4 + 1
+    ASSERT 0 = w3*w5
+    ASSERT w5 = w2
+
+    unconstrained func 0: directive_invert
+    0: @21 = const u32 1
+    1: @20 = const u32 0
+    2: @0 = calldata copy [@20; @21]
+    3: @2 = const field 0
+    4: @3 = field eq @0, @2
+    5: jump if @3 to 8
+    6: @1 = const field 1
+    7: @0 = field field_div @1, @0
+    8: stop &[@20; @21]
+    ");
+}
+
+#[test]
+fn unchecked_add_u8() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8, v1: u8):
+        v2 = unchecked_add v0, v1
+        return v2
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1]
+    public parameters: []
+    return values: [w2]
+    BLACKBOX::RANGE input: w0, bits: 8
+    BLACKBOX::RANGE input: w1, bits: 8
+    ASSERT w2 = w0 + w1
+    ");
+}
+
+#[test]
+fn checked_add_u8_no_predicate() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8, v1: u8):
+        v2 = add v0, v1
+        return v2
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1]
+    public parameters: []
+    return values: [w2]
+    BLACKBOX::RANGE input: w0, bits: 8
+    BLACKBOX::RANGE input: w1, bits: 8
+    ASSERT w3 = w0 + w1
+    BLACKBOX::RANGE input: w3, bits: 8
+    ASSERT w3 = w2
+    ");
+}
+
+#[test]
+fn checked_add_u8_with_predicate() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8, v1: u8, v2: u1):
+        enable_side_effects v2
+        v3 = add v0, v1
+        return v3
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // Note how every operand in the addition (w0, w1) is multiplied by the predicate (w2)
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1, w2]
+    public parameters: []
+    return values: [w3]
+    BLACKBOX::RANGE input: w0, bits: 8
+    BLACKBOX::RANGE input: w1, bits: 8
+    BLACKBOX::RANGE input: w2, bits: 1
+    ASSERT w4 = w0*w2 + w1*w2
+    BLACKBOX::RANGE input: w4, bits: 8
+    ASSERT w4 = w3
+    ");
+}
+
+#[test]
+fn checked_mul_u8_with_predicate() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8, v1: u8, v2: u1):
+        enable_side_effects v2
+        v3 = mul v0, v1
+        return v3
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // w0 is multiplied by w1, then the reslt (w4) is multiplied by the predicate (w2)
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1, w2]
+    public parameters: []
+    return values: [w3]
+    BLACKBOX::RANGE input: w0, bits: 8
+    BLACKBOX::RANGE input: w1, bits: 8
+    BLACKBOX::RANGE input: w2, bits: 1
+    ASSERT w4 = w0*w1
+    ASSERT w5 = w2*w4
+    BLACKBOX::RANGE input: w5, bits: 8
+    ASSERT w5 = w3
+    ");
+}
+
+#[test]
+fn div_u8_no_predicate_by_var() {
```

Comment:
```suggestion
fn div_u8_no_predicate_by_witness() {
```
Prefer this to `var`